python/ray/util/tracing.py

```diff
@@ -0,0 +1,400 @@
+    return _global_is_tracing_enabled
+
+
+class DictPropagator:
+    def inject_current_context() -> Dict[Any, Any]:
+        """Inject trace context into otel propagator."""
+        context_dict: Dict[Any, Any] = {}
+        propagators.inject(dict.__setitem__, context_dict)
+        return context_dict
+
+    def extract(context_dict: Dict[Any, Any]):
+        """Given a trace context, extract as a Context."""
+        return cast(Context, propagators.extract(DictGetter(), context_dict))
+
+
+@contextmanager
+def use_context(parent_context) -> Generator[None, None, None]:
+    """Uses the Ray trace context for the span."""
+    new_context = parent_context if parent_context is not None else Context()
+    token = context.attach(new_context)
+    try:
+        yield
+    finally:
+        context.detach(token)
+
+
+def _function_hydrate_span_args(func: Callable[..., Any]):
+    """Get the Attributes of the function that will be reported as attributes
+    in the trace."""
+    runtime_context = get_runtime_context().get()
+
+    span_args = {
+        "ray.remote": "function",
+        "ray.function": func,
+        "ray.pid": str(os.getpid()),
+        "ray.job_id": runtime_context["job_id"].hex(),
+        "ray.node_id": runtime_context["node_id"].hex(),
+    }
+
+    # We only get task ID for workers
+    if ray.worker.global_worker.mode == ray.worker.WORKER_MODE:
+        task_id = (runtime_context["task_id"].hex()
+                   if runtime_context["task_id"] else None)
+        if task_id:
+            span_args["ray.task_id"] = task_id
+
+    worker_id = getattr(ray.worker.global_worker, "worker_id", None)
+    if worker_id:
+        span_args["ray.worker_id"] = worker_id.hex()
+
+    return span_args
+
+
+def _function_span_producer_name(func: Callable[..., Any]) -> str:
+    """Returns the function span name that has span kind of producer."""
+    args = _function_hydrate_span_args(func)
+    assert args is not None
+    name = args["ray.function"]
+
+    return f"{name} ray.remote"
+
+
+def _function_span_consumer_name(func: Callable[..., Any]) -> str:
+    """Returns the function span name that has span kind of consumer."""
+    args = _function_hydrate_span_args(func)
+    assert args is not None
+    name = args["ray.function"]
+
+    return f"{name} ray.remote_worker"
+
+
+def _actor_hydrate_span_args(class_: _nameable, method: _nameable):
+    """Get the Attributes of the actor that will be reported as attributes
+    in the trace."""
+    if callable(class_):
+        class_ = class_.__name__
+    if callable(method):
+        method = method.__name__
+
+    runtime_context = get_runtime_context().get()
+
+    span_args = {
+        "ray.remote": "actor",
+        "ray.actor_class": class_,
+        "ray.actor_method": method,
+        "ray.function": f"{class_}.{method}",
+        "ray.pid": str(os.getpid()),
+        "ray.job_id": runtime_context["job_id"].hex(),
```

Comment:
random thought but it's kind of weird that we need to call `.hex()` here, this should probably just be `hex` to begin with...